datafusion/physical-plan/src/joins/cross_join.rs

```diff
@@ -253,9 +254,11 @@ impl ExecutionPlan for CrossJoinExec {
             schema: self.schema.clone(),
             left_fut,
             right: stream,
-            right_batch: Arc::new(parking_lot::Mutex::new(None)),
+            right_batch: RecordBatch::new_empty(self.right().schema()),
```

Comment:
Might be cleaner to store right batch in `state` (i.e. in `BuildBatches`)

Reply:
Done, thanks